navapbc/template-application-nextjs
# Setup i18n for next.js, jest, and storybook

## Ticket

- https://wicmtdp.atlassian.net/browse/WMDP-147

## Changes
> What was added, updated, or removed in this PR.

- Install and configure [next-i18next](https://github.com/i18next/next-i18next) for Next.js internationalization
- Move next.js i18n config out of `next.config.js` and into `next-i18next.config.js`
- Modify `pages/_app.tsx` and `pages/index.tsx` to support i18n
- Move jest i18n config into `/tests/jest-i18n.ts`
- Install and configure [storybook-react-i18next](https://storybook.js.org/addons/storybook-react-i18next) for storybook internationalization
- Add support for `<em>` tags in react-i18next in both next.js and storybook.

### Extras
- Fix `space-before-function-paren` eslint rule so that it allows spaces before parens as listed in the README.md. Matches what prettier does
- Change prettier so that it removes `trailingComma`s. Matches what eslint expects
- Rename `test` dir to `tests` (plural)

## Context for reviewers
> Testing instructions, background context, more in-depth details of the implementation, and anything else you'd like to call out or ask reviewers. Explain how the changes were verified.

This PR stacks on top of #15 (branch https://github.com/navapbc/template-application-nextjs/tree/rocket/wmdp-36-storybook).

NOTE: NextJS will automatically detect with local the user preders based on the 'Accept-Language' header and the current domain. User will be redirected to detected locale's subpath -- i.e. 'example.com/es'

NOTE: This PR extends and brings over work from #18

## Testing
> Screenshots, GIF demos, code examples or output to help show the changes working as expected. ProTip: you can drag and drop or paste images into this textbox.

1. Run `yarn install` and `yarn dev` to assert page renders correctly with translated text
2. Navigate to `localhost:3000/es` to view spanish translation
3. Run `yarn storybook` to assert stories render correctly with translated text
4. Click the globe and select "Español" to view the spanish translation
5. Run `yarn jest` to assert tests pass
